internal/pkg/template/templates/environment/partials/cdn-resources.yml

```diff
@@ -1,10 +1,53 @@
+UniqueAliasesFunctionRole:
+  Metadata:
+    'aws:copilot:description': 'An IAM Role for the unique aliases lambda'
+  Type: AWS::IAM::Role
+  Properties:
+    AssumeRolePolicyDocument:
+      Version: '2012-10-17'
+      Statement:
+        - Effect: Allow
+          Principal:
+            Service:
+              - lambda.amazonaws.com
+          Action:
+            - sts:AssumeRole
+    Path: /
+    ManagedPolicyArns:
+      - !Sub arn:${AWS::Partition}:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole
+
+UniqueAliasesFunction:
+  Type: AWS::Lambda::Function
+  Condition: CreateALB
+  Properties:
+    {{- with $cr := index .CustomResources "UniqueAliasesFunction" }}
+    Code:
+      S3Bucket: {{$cr.Bucket}}
+      S3Key: {{$cr.Key}}
+    {{- end}}
+    Handler: "index.handler"
+    Timeout: 600
+    MemorySize: 512
+    Role: !GetAtt 'UniqueAliasesFunctionRole.Arn'
+    Runtime: nodejs12.x
+
+UniqueAliasesAction:
+  Metadata:
+    'aws:copilot:description': 'An action to get unique custom aliases for services in your environment'
+  Condition: CreateALB
+  Type: Custom::UniqueAliasesFunction
+  Properties:
+    ServiceToken: !GetAtt UniqueAliasesFunction.Arn
+    Aliases: !Ref Aliases
+
 CloudFrontDistribution:
   Metadata:
     'aws:copilot:description': 'A CloudFront distribution for global content delivery'
   Condition: CreateALB
   Type: AWS::CloudFront::Distribution
   Properties:
     DistributionConfig:
+      Aliases: !GetAtt UniqueAliasesAction.UniqueAliases
```

Comment:
Does the distribution have a configuration error when there are Aliases but no SSL/TLS certificate? I can test this and get back to you.

Reply:
Yes, this will error out if the user has working HTTPS and then tries to remove it. This happens because they will have no SSL/TLS Certificate, but since the `Aliases` parameter of the env stack isn't removed until they redeploy the service.

This leaves the user in deadlock because the env can't be redeployed until all `Aliases` are removed, but the services can't be redeployed while the imported certificates still require those `Aliases`.

You can just move the `Aliases` field into the same conditions as `ViewerCertificate` as it needs to be filled out in the exact same conditions 👍

Reply:
Does it work when we don't need to specify any alias?

Reply:
ok i changed the condition to only set `Aliases` if one of these is true:
- they created the app with `--domain`
- they imported certs to the cdn